### PR TITLE
fix: refresh suggestion cache entry on derivations update

### PR DESCRIPTION
**Pull request by erictapen**
Couldn't sleep with having this problem in my head.

I had to switch the arguments for the update_or_create function because they were wrong: The proposal_id is used for the lookup and the payload is then used on update.

This is going to make updates very slow, but doesn't affect page load and I'd argue that it is important to have correct behaviour at first. With https://github.com/Nix-Security-WG/nix-security-tracker/pull/383 merged update times wouldn't even be noticable.

### CODE REVIEW

File: src/website/webview/views.py [hunk 1]
```diff
@@ -545,7 +547,10 @@ def post(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
         elif new_status == "ACCEPTED":
             suggestion.status = CVEDerivationClusterProposal.Status.ACCEPTED
 
-        suggestion.save()
+        with transaction.atomic():
+            suggestion.save()
+            cache_new_suggestions(suggestion)
```
Comment (fricklerhandwerk) [inline src/website/webview/views.py:552]: This should be done triggered async, otherwise we'll have to wait a long time.

**Comment by RaitoBezarius** [comment]
I looked into that more seriously, I don't think triggering a recaching is the good approach here.

For one, this will trigger a weird experience where you will see temporarily (in case of heavy loads) the view non-sync with what's going on.

Recaching should be reserved for expensive transformations that are hard to re-express on their cached variants.

In this instance, there's only one operation: filtering out derivations from the payload.

This should be expressible in one SQL query which should not involve back'n'forth with Python. I have some local prototype I'm working on where I can do this sort of stuff.

**Comment by fricklerhandwerk** [comment]
Are you suggesting something like storing a flag for the filtered-out derivations, but somewhere outside the actual suggestion entry (since that is cached)?

**Comment by RaitoBezarius** [comment]
This is now ready.

**Comment by RaitoBezarius** [comment]
Rebased.

**Comment by RaitoBezarius** [comment]
The rebase broke displaying derivations.

**Comment by RaitoBezarius** [comment]
OK this is beyond my paygrade, I do see the derivations threaded in some templates but I don't understand what's going on, can you weigh in @erictapen (whenever you are available ofc)?

**Comment by RaitoBezarius** [comment]
In the meantime, fix #396 as intended in principle.

**Comment by fricklerhandwerk** [comment]
This also breaks state change display, setting it to draft for now.

**Comment by erictapen** [comment]
Fixed!

**Review by fricklerhandwerk — APPROVED** [review]
Tested, works. Displaying pages in the "selected" view for some reason is much slower than in the "suggestions" view, which doesn't make much sense to me, but we'll deal with that another time.

**Pull request merged**